website/content/docs/k8s/installation/vault/data-integration/webhook-certs.mdx

```diff
@@ -0,0 +1,261 @@
+  - controller and connect inject each locally update their mutatingwebhookconfigurations so that Kubernetes can relay events.
+  - Vault will manage certificate rotation and each of these webhooks will seamlessly received rotated certificates.
+
+To use Vault as the controller and connect inject Webhook Certificate Provider, we will need to modify the steps outlined in the [Data Integration](/docs/k8s/installation/vault/data-integration) section:
+
+### One time setup in Vault
+  1. Create a Vault policy that authorizes the desired level of access to the secret.
+
+### Setup per Consul datacenter
+  1. (Added) Create Vault PKI roles for controller and connect inject each that establish the domains that each is allowed to issue certificates for.
+  1. Create Vault Kubernetes auth roles that link the policy to each Consul on Kubernetes service account that requires access.
+  1. Configure the Vault Kubernetes auth roles in the Consul on Kubernetes helm chart.
+
+## One time setup in Vault
+
+### Create a Vault policy that authorizes the desired level of access to the secret
+To use Vault to issue controller or connect inject webhook certificates, you will need to create the Vault Policies that will allow either controller or connect inject to access its respective certificate issuing url.
+
+#### Create Vault Policies for the Controller and Connect Inject Webhook Certificates
+
+-> **Note:** The PKI secret paths referenced by the Vault Policies below will be your `global.secretsBackend.vault.controller.tlsCert.secretName` and `global.secretsBackend.vault.connectInject.tlsCert.secretName` Helm values respectively.
+
+Next, we will create a policy that allows `["create", "update"]` access to the
+[certificate issuing URL](https://www.vaultproject.io/api/secret/pki#generate-certificate) so Consul controller and connect inject can
+fetch a new certificate/key pair and provide it to the Kubernetes mutatingwebhookconfiguration.
+
+
+```shell-session
+$ vault policy write controller-tls-policy - <<EOF
+path controller/issue/controller-role {
+  capabilities = ["create", "update"]
+}
+EOF
+```
+
+```shell-session
+$ vault policy write connect-inject-policy - <<EOF
+path connect-inject/issue/connect-inject-role {
+  capabilities = ["create", "update"]
+}
+EOF
+```
+#### Create Vault Policies for the CA URL
+
+-> **Note:** The PKI secret paths referenced by the Vault Policies below will be your `global.secretsBackend.vault.controller.caCert.secretName` and `global.secretsBackend.vault.connectInject.caCert.secretName` Helm values respectively.
+
+Next, we will create a policy that allows `["read"]` access to the [CA URL](https://www.vaultproject.io/api/secret/pki#read-certificate),
+this is required for the Consul components to communicate with the Consul servers in order to fetch their auto-encryption certificates.
```

Comment:
```suggestion
Next, create a policy that allows `["read"]` access to the [CA URL](https://www.vaultproject.io/api/secret/pki#read-certificate). The policy is required so that Consul components can communicate with the Consul servers in order to fetch their auto-encryption certificates. Issue the following commands to create the policy:
```